ui-charts/src/spaceTimeChart/components/Quadrilateral.tsx

```diff
@@ -0,0 +1,52 @@
+import { useCallback } from 'react';
+
+import { useDraw } from '../hooks/useCanvas';
+import { type DataPoint, type DrawingFunction } from '../lib/types';
+
+export type QuadrilateralProps = {
+  vertices: [DataPoint, DataPoint, DataPoint, DataPoint];
+  style: {
+    backgroundColor: string;
+    borderColor: string;
+    borderWidth: number;
+  };
+};
+
+/**
+ * Draws a quadrilateral in the space time chart
+ * Position correspond to each point of the quadrilateral
+ *
+ *      vertices[0]   ________ vertices[1]
+ *                  /        /
+ *     vertices[3] /________/ vertices[2]
+ *
+ */
+export const Quadrilateral = ({ vertices, style }: QuadrilateralProps) => {
+  const drawRegion = useCallback<DrawingFunction>(
+    (ctx, { getSpacePixel, getTimePixel }) => {
+      ctx.save();
+
+      ctx.fillStyle = style.backgroundColor ?? 'lightblue';
+      ctx.strokeStyle = style.borderColor ?? 'lightblue';
+      ctx.lineWidth = style.borderWidth ?? 1;
+
+      ctx.beginPath();
+      vertices.forEach((dataPoint, index) => {
+        if (index === 0) {
+          ctx.moveTo(getTimePixel(dataPoint.time), getSpacePixel(dataPoint.position));
+        } else {
+          ctx.lineTo(getTimePixel(dataPoint.time), getSpacePixel(dataPoint.position));
+        }
+      });
```

Comment:
```suggestion
      vertices.forEach((dataPoint) => {
          ctx.lineTo(getTimePixel(dataPoint.time), getSpacePixel(dataPoint.position));
      });
```

nit: you draw only one shape, no need to seek for first index here

Reply:
Have you tested this? I think we need to move the canvas' current point, before we can start drawing lines?

Reply:
Can’t this cause unexected behavior ?

Reply:
I also though it was mandatory 🤔, but I just tested it and it seems to work fine

Reply:
> Have you tested this? I think we need to move the canvas' current point, before we can start drawing lines?

indeed
it seems moveTo and lineTo are the same, expect that lineTo draws, if it's the first point nothing will be draw

Reply:
Right, just read the spec and it seems like `lineTo()` works just like `moveTo()` when there is no sub-path: https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-lineto